src/compound_model.js

```diff
@@ -0,0 +1,165 @@
+import { InMemoryEntity } from "@exabyte-io/code.js/dist/entity";
+import { setNextLinks, setUnitsHead } from "@exabyte-io/code.js/dist/utils";
+
+import { UnitModel } from "./unit_model";
+import { UnitModelFactory } from "./unit_model_factory";
+import { buildModelPath } from "./utils";
+
+export class Model extends InMemoryEntity {
+    static UnitModelFactory = UnitModelFactory;
+
+    static UnitModelCls = UnitModel;
+
+    constructor({ ...config } = {}) {
+        super(config);
+        this._units = Model.instantiateUnits(config.units);
+        this.setNextLinks().setUnitsHead().syncToProp();
+    }
+
+    get name() {
+        return this.prop("name");
+    }
+
+    get slug() {
```

Comment:
Don't we already have `get slug` inside `InMemoryEntity`?

Reply:
yes you are right, I'll delete this function 👍